client/netdb.c

```diff
@@ -158,20 +160,23 @@ static struct host *find_add_ipport(struct host *h, const char *ipport, int flag
 
 static int read_database(const char *fname, int flags)
 {
-	uint32_t ips[MAX_BLOCKED_IPS * MAX_ALLOWED_FROM_IP];
-	uint8_t ips_count[MAX_BLOCKED_IPS * MAX_ALLOWED_FROM_IP];
+	uint32_t ips[MAX_BLOCKED_IPS * MAX_ALLOWED_FROM_IP] = {0};
+	uint8_t ips_count[MAX_BLOCKED_IPS * MAX_ALLOWED_FROM_IP] = {0};
 	struct host h0;
-	char str[64], *p;
+	char str[64] = {0}, *p = NULL;
 	FILE *f = xdag_open_file(fname, "r");
 	int n = 0, n_ips = 0, n_blocked = 0, i;
 
 	if (!f) return -1;
 
 	while (fscanf(f, "%63s", str) == 1) {
-		p = strchr(str, ':');
-		if (!p || !p[1]) continue;
-	
-		h0.flags = 0;
+		p = strrchr(str, ':');
```

Comment:
why strrchr instead of strchr?

Reply:
for most string like this
xxx.xxx.xxx.xxx:port
I think using strrchr is better than strchr

Reply:
to search the last :, you have first to find the end of the string, so it will be slower